Dockerfile

```diff
@@ -34,30 +40,67 @@ RUN apt-get -y update && \
         libxpm-dev \
         make \
         openssl \
+        python3 \
+        python3-pip \
         python-dev \
         python-pip \
         xlibmesa-glu-dev \
         zlib1g-dev && \
-    rm -rf /var/lib/apt/lists/* && \
-    apt-get -y autoremove && \
-    apt-get -y clean
-
-ENV ROOTSYS /usr/local
-RUN git clone --quiet --depth 1 --branch v6-18-04 http://root.cern.ch/git/root.git /code/root && \
-    cd /code && \
-    mkdir _build && \
-    cd _build && \
-    cmake -Dcxx17=On -Dbuiltin_xrootd=On ../root && \
-    cmake --build . -- -j3 && \
-    cmake -P cmake_install.cmake && \
-    cd / && \
-    rm -rf /code
+    yum clean all && \
+    rm -rf /var/cache/yum
+
+RUN pip install --upgrade pip
+
+# gfal2-python bindings
+RUN curl -o /etc/yum.repos.d/gfal2-repo.repo https://dmc-repo.web.cern.ch/dmc-repo/dmc-el7.repo && \
+    git clone --branch v1.12.0 https://github.com/cern-fts/gfal2-python.git && \
+    cd gfal2-python/ && \
+    ./ci/fedora-packages.sh && \
+    cd packaging/ && \
+    RPMBUILD_SRC_EXTRA_FLAGS="--without docs --without python2" make srpm && \
+    yum-builddep -y python3-gfal2 && \
+    pip install gfal2-python
+
+RUN pip3 install rucio-clients==1.29.1
+
+# EGI trust anchors
+RUN curl -Lo /etc/yum.repos.d/EGI-trustanchors.repo https://repository.egi.eu/sw/production/cas/1/current/repo-files/EGI-trustanchors.repo && \
+    yum -y install ca-certificates ca-policy-egi-core && \
+    yum -y update
+
+# make cmake3 the default version of cmake
+RUN alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake 10 \
+    --slave /usr/local/bin/ctest ctest /usr/bin/ctest \
+    --slave /usr/local/bin/cpack cpack /usr/bin/cpack \
+    --slave /usr/local/bin/ccmake ccmake /usr/bin/ccmake \
+    --family cmake
+RUN alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake3 20 \
+    --slave /usr/local/bin/ctest ctest /usr/bin/ctest3 \
+    --slave /usr/local/bin/cpack cpack /usr/bin/cpack3 \
+    --slave /usr/local/bin/ccmake ccmake /usr/bin/ccmake3 \
+    --family cmake
+
+# ROOT
+RUN yum install -y gcc-c++ \
+    libX11-devel libXpm-devel libXft-devel libXext-devel python openssl-devel
+RUN git clone --branch v6-24-00-patches --depth=1 https://github.com/root-project/root.git root_src
+RUN mkdir root_build root_install
+WORKDIR root_build
+RUN cmake -DCMAKE_INSTALL_PREFIX=../root_install ../root_src
+RUN cmake --build . -- install -j4
+RUN source ../root_install/bin/thisroot.sh
```

Comment:
Can you clean up after the build any unnecessary directories etc? Note that we do `rm -rf /code` in `Dockerfile` in the `master` branch.  Otherwise, without appropriate cleaning, the image is much bigger than it needs to be:

```console
$ docker images | grep root6 | grep 6 | sed 's, \+, ,g'
reanahub/reana-env-root6 v6-24-00-patches 2afaab937438 13 minutes ago 4.65GB
reanahub/reana-env-root6 6.18.04 c754f13f32ae 2 years ago 2.07GB
```